docs/auto-tag-orders-with-utm-parameters/script.liquid

```diff
@@ -112,22 +126,44 @@
 {% elsif event.topic == "mechanic/shopify/bulk_operation" %}
   {% if event.preview %}
     {% capture bulkOperation_objects_json %}
-      {"__typename":"Order","id":"gid:\/\/shopify\/Order\/1234567890","name":"#1234","tags":[],"customerJourney":{"moments":[{"utmParameters":{"campaign":"Google Shopping","content":"shoppingcontent","medium":"cpc","source":"google","term":"shoppingterm"}}]}}
+      {"__typename":"Order","id":"gid:\/\/shopify\/Order\/1234567890","name":"#1234","tags":[],"customerJourneySummary":{}}
+      {"__typename":"CustomerVisit","utmParameters":{"campaign":"New Campaign 001","content":null,"medium":"email","source":"newsletter","term":null},"__parentId":"gid:\/\/shopify\/Order\/1234567890"}
+      {"__typename":"CustomerVisit","utmParameters":{"campaign":"New Campaign 001","content":"cta_link","medium":"email","source":"drip","term":"automation+shopify"},"__parentId":"gid:\/\/shopify\/Order\/1234567890"}
     {% endcapture %}
 
     {% assign bulkOperation = hash %}
     {% assign bulkOperation["objects"] = bulkOperation_objects_json | parse_jsonl %}
   {% endif %}
 
-  {% assign orders = bulkOperation.objects %}
```

Comment:
I thiiiink we can skip the entire new forloop structure, if we prepare for the parameter set work like so:

```liquid
{% assign orders = bulkOperation.objects | where: "__typename", "Order" %}

{% for order in orders %}
  {% assign parameter_sets = bulkOperation.objects | where: "__typename", "CustomerVisit" | where: "__parentId", order.id | map: "utmParameters"%}

  {% for parameter_set in parameter_sets %}
    ... and we're back to the original code
```

Does that make sense? Does that actually work, in practice?

Reply:
Oh, I like this approach much better. I'll try it out and report back.

Reply:
Works great, had to put a conditional in to handle the edges/node on the orders/create query though.

Reply:
Wow, yeah, this looks super nice. Good work!